### PR TITLE
VxDesign: Add custom election title field to precinct split form

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

Changing the solution for #4369. In #4482, we hardcoded an election title override for NH school district elections. However, we have more customizations needed for school district elections (e.g. adding a clerk signature: #4481). So we'll need a way to attach custom content to precinct splits (which is how we'll model the town and school elections - each will have its own precinct split).

This PR removes the election title override hardcoding and instead adds a field for a custom election title override to the precinct split form.

## Demo Video or Screenshot

https://github.com/votingworks/vxsuite/assets/530106/285f9b46-c4e0-49a5-9860-330d6870ccfa

## Testing Plan
- Updated existing tests
- Added new tests to cover the geography screen
- Some basic manual testing

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: apps/design/backend/src/store.ts [hunk 1]
```diff
@@ -213,6 +219,42 @@ export function generateBallotStyles(
   }
 }
 
+export function assignCustomContentToBallotStyles(
```
Comment (eventualbuddha) [inline apps/design/backend/src/store.ts:222]: This name sounds like it mutates something. Maybe call it `getNhCustomContentByBallotStyle`?

File: apps/design/frontend/src/geography_screen.tsx [hunk 2]
```diff
@@ -406,10 +406,10 @@ function PrecinctForm({
   precinctId?: PrecinctId;
   savedPrecincts: Precinct[];
   districts: readonly District[];
-}): JSX.Element {
-  const [precinct, setPrecinct] = useState<Precinct>(
+}): JSX.Element | null {
+  const [precinct, setPrecinct] = useState<Precinct | undefined>(
```
Comment (eventualbuddha) [inline apps/design/frontend/src/geography_screen.tsx:418]: You can just infer the type parameter:

```suggestion
  const [precinct, setPrecinct] = useState(
```

**Review by eventualbuddha — APPROVED** [review]

File: apps/design/frontend/src/geography_screen.tsx [hunk 3]
```diff
@@ -406,10 +414,10 @@ function PrecinctForm({
...
     precinctId
-      ? find(savedPrecincts, (p) => p.id === precinctId)
+      ? savedPrecincts.find((p) => p.id === precinctId)
       : // To make mocked IDs predictable in tests, we pass a function here
         // so it will only be called on intial render.
```
Comment (eventualbuddha) [inline apps/design/frontend/src/geography_screen.tsx:422]: If you make any other changes to this PR, maybe also correct the spelling of `initial`?

File: apps/design/frontend/src/geography_screen.tsx (hunk 2 above)
Comment (jonahkagan) [reply apps/design/frontend/src/geography_screen.tsx:418]: Huh, I thought I changed it in response to an error but it seems to work now 🤷

File: apps/design/frontend/src/geography_screen.tsx (hunk 3 above)
Comment (jonahkagan) [reply apps/design/frontend/src/geography_screen.tsx]: Fixed, thanks

**Pull request merged**